WebApp/src/main/java/com/isa/webapp/controller/TeacherController.java

```diff
@@ -1,115 +1,80 @@
 package com.isa.webapp.controller;
 
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.isa.webapp.model.GradeForm;
-import com.isa.webapp.model.Subjects;
-import com.isa.webapp.model.User;
-import com.isa.webapp.model.UserRole;
-import com.isa.webapp.service.StudentService;
-import org.springframework.beans.factory.annotation.Autowired;
+import com.isa.webapp.model.*;
+import com.isa.webapp.repository.GradeRepository;
+import com.isa.webapp.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
 import org.springframework.security.core.annotation.AuthenticationPrincipal;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.ModelAttribute;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.*;
 
 import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.ArrayList;
+import java.time.LocalDate;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 
 @Controller
+@RequiredArgsConstructor
 public class TeacherController {
 
-    private final StudentService studentService;
+    private final UserRepository userRepository;
```

Comment:
wolałbym, żeby tu była zależność do Serwisu, który wewnątrz korzysta z repozyrium
tak żeby flow wywołań zawsze było takie: controller -> service -> repository

Reply:
Maciej, możesz pokazać w jaki sposób mogę zaimplementować controller -> service -> repository na tym przykładzie?

Reply:
np. na przykładzie tego controllera i metody `showStudentList` to ja bym zrobił to tak:
```
public class TeacherController {

    private final UserService userService;

    @GetMapping("/teacher/students")
    public String showStudentList(Model model, @AuthenticationPrincipal UserDetails userDetails) {
        model.addAttribute("students", userService.getStudentsByRole(UserRole.STUDENT));
        model.addAttribute("role", userService.getUserRole(userDetails)); // tutaj zamiast zwracania isTeacher, albo isStudent to dlaczego by nie zwrócić po prostu roli, ale jak chcesz to może być i isTeacher, ale jako metoda wydzielona do serwisu -> userService.isTeacherRole(userDetails)
        model.addAttribute("username", userService.getUsername(userDetails));
        return "teacher_student_list";
    }
}
```

Reply:
chodzi głównie o to, żeby controllery nie miały dostępu do repozytoriów, tylko do serwisów oraz, żeby metody w controllerach był jak najmniejsze, np. wywołując od razu metody serwisowe przy atrybutach